pkg/rehearse/jobs_test.go

```diff
@@ -254,38 +255,175 @@ func makeTestingProwJob(name, namespace, jobName, context string, refs *pjapi.Re
 	}
 }
 
-func TestExecuteJobs(t *testing.T) {
-	testLogger := logrus.New()
+func makeTestData() (int, string, string, *pjapi.Refs) {
 	testPrNumber := 123
 	testNamespace := "test-namespace"
-	testRepo := "testRepo"
-	testOrg := "testOrg"
 	testRefs := &pjapi.Refs{
-		Org:     testOrg,
-		Repo:    testRepo,
+		Org:     "testRepo",
+		Repo:    "testOrg",
 		BaseRef: "testBaseRef",
 		BaseSHA: "testBaseSHA",
 		Pulls:   []pjapi.Pull{{Number: testPrNumber, Author: "testAuthor", SHA: "testPrSHA"}},
 	}
+	testReleasePath := "path/to/openshift/release"
+
+	return testPrNumber, testNamespace, testReleasePath, testRefs
+}
+
+func makeSuccessfulFinishReactor(watcher watch.Interface, jobs map[string][]prowconfig.Presubmit) func(clientgo_testing.Action) (bool, watch.Interface, error) {
+	return func(clientgo_testing.Action) (bool, watch.Interface, error) {
+		watcher.Stop()
+		n := 0
+		for _, jobs := range jobs {
+			n += len(jobs)
+		}
+		ret := watch.NewFakeWithChanSize(n, true)
+		for event := range watcher.ResultChan() {
+			pj := event.Object.(*pjapi.ProwJob).DeepCopy()
+			pj.Status.State = pjapi.SuccessState
+			ret.Modify(pj)
+		}
+		return true, ret, nil
+	}
+}
+
+func TestExecuteJobsErrors(t *testing.T) {
+	testPrNumber, testNamespace, testRepoPath, testRefs := makeTestData()
+	targetRepo := "targetOrg/targetRepo"
+
+	testCases := []struct {
+		description string
+		jobs        map[string][]prowconfig.Presubmit
+		reactor     func(action clientgo_testing.Action) (handled bool, ret runtime.Object, err error)
```

Comment:
You might be able to DRY this out to a list of job names that could fail on create and make the test cases more narrow